tracing-appender/src/rolling/builder.rs

```diff
@@ -181,6 +183,34 @@ impl Builder {
         Self { suffix, ..self }
     }
 
+    /// Keep the last `n` log files on disk.
+    ///
+    /// When a new log file is created, if there are `n` or more
+    /// existing log files in the directory, the oldest will be deleted.
+    /// If no value is supplied, `RollingAppender` will not remove any files.
+    ///
```

Comment:
It occurs to me that we might also want to describe the rules that we use to determine whether or not a file is something we can delete. Maybe something like this:
```suggestion
    /// 
    /// Files are considered candidates for deletion based on the following
    /// criteria:
    ///
    /// * The file must not be a directory or symbolic link.
    /// * If the appender is configured with a [`filename_prefix`], the file
    ///   name must start with that prefix.
    /// * If the appender is configured with a [`filename_suffix`], the file
    ///   name must end with that suffix.
    /// * If the appender has neither a filename prefix nor a suffix, then the
    ///   file name must parse as a valid date based on the appender's date
    ///   format.
    ///
    /// Files matching these criteria may be deleted if the maximum number of
    /// log files in the directory has been reached.
```
this way, users should hopefully know not to store other files created by other programs that _happen_ to match the deletion criteria in the same directory.